#! /bin/sh /usr/share/dpatch/dpatch-run
## 22_fix_12_year_display.dpatch by  <jblache@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix display when both -12 and -year are given. Fix for 228985.

@DPATCH@
diff -urNad wmclock-1.0.12.2~/wmclock.c wmclock-1.0.12.2/wmclock.c
--- wmclock-1.0.12.2~/wmclock.c	2008-11-16 18:23:07.185633821 +0100
+++ wmclock-1.0.12.2/wmclock.c	2008-11-16 18:23:25.482650412 +0100
@@ -828,7 +828,14 @@
 #endif /* !ONLY_SHAPED_WINDOW */
    for (i = 0; i < NUM_TIME_POSITIONS; i++)
     {
-       xPos[i] += enable12HourClock ? timePos24[i] : timePos12[i];
+      if (enable12HourClock && (!enableYearDisplay))
+       {
+         xPos[i] += timePos24[i];
+       }
+      else
+       {
+         xPos[i] += timePos12[i];
+       }
     }
    
    /* Open the display */
